File: UiMobile/lets_play/lib/model/event.dart
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:lets_play/model/category.dart';
import 'package:lets_play/model/city.dart';
import 'package:lets_play/model/user.dart';

class Event {
  int id;
  String title;
  User createdByUser;
  DateTime startDate;
  String exactAddress;
  String description;
  DateTime createdAt;
  City city;
  int peopleNeeded;
  DateTime updatedAt;
  String status;
  bool isActive;
  Category category;
  int recurring;

  Event(
      {@required this.id,
      @required this.title,
      @required this.createdByUser,
      @required this.startDate,
      @required this.description,
      @required this.createdAt,
      @required this.city,
      @required this.peopleNeeded,
      @required this.updatedAt,
      @required this.status,
      @required this.isActive,
      @required this.category,
      @required this.exactAddress,
      @required this.recurring});

  Event.fromJson(Map<String, dynamic> json)
      : title = json['title'] as String,
        startDate = DateTime.parse(json['eventStartDate']),
        id = json['id'] as int,
        createdByUser = User.fromJson(json['user']),
        createdAt = DateTime.parse(json['createdOn']),
        description = json['description'] as String,
        city = City.fromJson(json['city']),
        peopleNeeded = json['peopleNeeded'] as int,
        status = json['status'],
        isActive = json['isactive'] as bool,
        category = Category.fromJson(json['category']),
        exactAddress = json['address'] as String,
        recurring = json['recurrnig'] as int;
}

File: DBService/DatabaseAPI/Models/DbModels/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace DatabaseAPI.Models.DbModels
{
    public enum Status
    {


        Upcoming = 1,
        HappeningNow = 2,
        Repeating = 3,
        Finished = 4,


    }
    public enum Recurring
    {

        EveryDay = 1,
        EveryWeek = 2,
        EveryMonth = 3,


    }
    public class Event
    {
        [Key]
        public int ID { get; set; }
        [Required]
        public DateTime EventStartDate { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedOn { get; set; } = DateTime.UtcNow;
        public Status EventStatus { get; set; }
        public Recurring? Recurring { get; set; }
        public string Description { get; set; }
        public int PeopleNeeded { get; set; }
        [Required]
        public string Title { get; set; }
        [Required]
        [ForeignKey("User")]
        public int UserCreatedByID { get; set; }
        public User User { get; set; }
        [Required]
        public Category Category { get; set; }
        [Required]
        public City City { get; set; }
        [Required]
        public string Address { get; set; }
        public ICollection<Comment> Comments { get; set; }
        public ICollection<UserEvent> Participants{ get; set; }








    }
}
